src/operator/tensor/util/node_op_util.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file node_op_util.h
+ * \brief abstraction for commonly nnvm::Node operations.
+ */
+#ifndef MXNET_OPERATOR_TENSOR_UTIL_NODE_OP_UTIL_H_
+#define MXNET_OPERATOR_TENSOR_UTIL_NODE_OP_UTIL_H_
+#include <mxnet/base.h>
+#include <string>
+#include <unordered_map>
+#include "../../elemwise_op_common.h"
+
+namespace mxnet {
+namespace util {
+
+class NodeOp {
```

Comment:
@larroy @apeforest ,
Have added this class as an abstraction over `MakeNode`. Using this we can avoid #15331 , as missing argument will be detected at compile time. Also it reduces the noise while using `MakeNode`. There shouldn't be any performance hit as the member functions should get inlined. 

`arctan` and `arctanh` both compute roughly the same thing. Can see the difference by using this class as abstraction and the our usual method. `arctan` -> Usual :: `arctanh` -> NodeOp

It might be possible that I have missed something. 

Let me know what you think, will move forward from there or reset the commit. 
Thanks.

Reply:
Loos good to me, indeed reduces noise.

Reply:
In that case, I'll also update `arctan` to use this. 
Thanks.

Reply:
@apeforest Do review this part. If it looks good, I'll update all other PRs to use this machinery once this is in.